src/utils/recommendations/preprocessing/index.js

```diff
@@ -0,0 +1,23 @@
+const tokenize = require('./tokenizer');
+const stem = require('./stemming');
+const lemmatize = require('./lemmatizer');
+const removeStopWords = require('./stopWords');
+
+function preprocessText(text) {
+  text = text
+    .toLowerCase()
+    .replace(/[^\w\s]/g, '')
+    .trim();
+
+  let tokens = tokenize(text);
+
+  tokens = removeStopWords(tokens);
+
+  tokens = lemmatize(tokens);
```

Comment:
My bad, we have to do either lemmatization or stemming & not both. Let's use Lemmatization in our case. You can lookup lemmatizing vs stemming for more info

Reply:
Keep the stemming util, just remove it from the index